Makefile: fix install rule

Do not link the .so with an absolute path, otherwise it may point to
the host library.

Based on the former patch by Yann E. MORIN.

Signed-off-by: Vicente Olivert Riera <vincent.riera@imgtec.com>
[baruch: update for 1.6]
Signed-off-by: Baruch Siach <baruch@tkos.co.il>

--- keyutils-1.5.9/Makefile.orig	2014-09-22 16:13:41.593562765 +0100
+++ keyutils-1.5.9/Makefile	2014-09-22 16:14:05.377963952 +0100
@@ -200,7 +200,7 @@ ifeq ($(NO_SOLIB),0)
 	$(INSTALL) -D $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	$(LNS) $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	mkdir -p $(DESTDIR)$(USRLIBDIR)
-	$(LNS) $(LIBDIR)/$(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
+	$(LNS) $(SONAME) $(DESTDIR)$(USRLIBDIR)/$(DEVELLIB)
 	sed \
 	-e 's,@VERSION\@,$(VERSION),g' \
 	-e 's,@prefix\@,$(PREFIX),g' \
